sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -396,6 +417,93 @@ private[spark] object HiveUtils extends Logging {
         config = configurations,
         barrierPrefixes = hiveMetastoreBarrierPrefixes,
         sharedPrefixes = hiveMetastoreSharedPrefixes)
+    } else if (hiveMetastoreJars == "path") {
+
+      def addLocalHiveJars(file: File): Seq[URL] = {
+        if (file.getName == "*") {
+          val files = file.getParentFile.listFiles()
+          if (files == null) {
+            logWarning(s"Hive jar path '${file.getPath}' does not exist.")
+            Nil
+          } else {
+            files.filter(_.getName.toLowerCase(Locale.ROOT).endsWith(".jar")).map(_.toURL).toSeq
+          }
+        } else {
+          file.toURL :: Nil
+        }
+      }
+
+      def checkRemoteHiveJars(path: String): Seq[URL] = {
+        try {
+          val hadoopPath = new Path(path)
+          val fs = hadoopPath.getFileSystem(hadoopConf)
+          if (hadoopPath.getName == "*") {
+            val parent = hadoopPath.getParent
+            if (!fs.exists(parent)) {
+              logWarning(s"Hive Jar ${path} does not exist.")
+              Nil
+            } else if (!fs.getFileStatus(parent).isDirectory) {
+              logWarning(s"Hive Jar ${parent} is not a directory.")
+              Nil
+            } else {
+              fs.listStatus(parent).map(_.getPath.toUri.toURL)
```

Comment:
Do we handle nested `*`, e.g., `path = `/foo/*/*`?

Reply:
> Do we handle nested `*`, e.g., `path = `/foo/_/_`?

you mean `path=/xxx/xxx/*/*` ?

Reply:
yes (sorry for the syntax).

Reply:
> yes (sorry for the syntax).

Not yet.

Reply:
Okay. Maybe it is worth pointing out in the description that we only support one single wildcard in the last path component.

Reply:
> Okay. Maybe it is worth pointing out in the description that we only support one single wildcard in the last path component.

Yea, good suggestion, we should make it more clear in desc.